flax/linen/recurrent.py

```diff
@@ -246,14 +244,19 @@ def __call__(self, carry: Tuple[Array, Array],
     hidden_features = h.shape[-1]
 
     def _concat_dense(inputs: Array,
-                      params: Mapping[str, Tuple[Array, Array]],
-                      use_bias: bool = True) -> Array:
+                      params: Mapping[str, Tuple[Array, Optional[Array]]],
+                      use_bias: bool = True) -> Dict[str, Array]:
       # Concatenates the individual kernels and biases, given in params, into a
       # single kernel and single bias for efficiency before applying them using
       # dot_general.
-      kernels, biases = zip(*params.values())
+      kernels = [kernel for kernel, _ in params.values()]
       kernel = jnp.concatenate(kernels, axis=-1)
       if use_bias:
+        biases = []
```

Comment:
This code is created to make `mypy` happy but also improves the correctness of the program.